packages/hardhat-core/src/internal/hardhat-network/provider/vm/dual.ts

```diff
@@ -13,11 +13,19 @@ import { HardhatBlockchainInterface } from "../types/HardhatBlockchainInterface"
 
 import { EthereumJSAdapter } from "./ethereumjs";
 import { RethnetAdapter } from "./rethnet";
-import { VMAdapter } from "./vm-adapter";
+import { Trace, VMAdapter } from "./vm-adapter";
 
 /* eslint-disable @nomiclabs/hardhat-internal-rules/only-hardhat-error */
 /* eslint-disable @typescript-eslint/restrict-template-expressions */
 
+function printEthereumJSTrace(trace: any) {
+  console.log(JSON.stringify(trace, null, 2));
+}
+
+function printRethnetTrace(_trace: any) {
+  // not implemented
```

Comment:
After returning _something_ as a trace in the rethnet adapter, you can implement how it will be printed here.

Depending on the value, the same implementation that `printEthereumJSTrace` uses might be enough.